libs/ngx-dsfr/password/src/lib/testing/password.component.spec.ts

```diff
@@ -0,0 +1,108 @@
+  let fixture: ComponentFixture<TestHostComponent>;
+  let testHost: TestHostComponent;
+  let componentUnderTest: DsfrPasswordComponent;
+  let harnessLoader: HarnessLoader;
+  let dsfrPasswordHarness: DsfrPasswordHarness;
+
+  const testLabel = 'testLabel';
+  const testId = 'testId';
+  const testPlaceholder = 'Enter your password...';
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [FormsModule],
+      declarations: [
+        DsfrPasswordComponent,
+        TestHostComponent
+      ]
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(TestHostComponent);
+    testHost = fixture.componentInstance;
+    componentUnderTest = fixture.debugElement.query(
+      By.directive(DsfrPasswordComponent)
+    ).componentInstance;
+
+    /*
+     * We're retrieving a HarnessLoader HERE to be able to get
+     * the Test Harness for the DsfrInputComponent LATER.
+     *
+     * We're NOT doing both HERE since getHarness() triggers
+     * indirectly the Change Detection mechanism and this,
+     * Data Binding and ngOnInit()...
+     */
+    harnessLoader = TestbedHarnessEnvironment.loader(fixture);
+  });
+
+  it('should be created', () => {
+    expect(componentUnderTest).toBeTruthy();
+  });
+
+  it('should throw an error when no id is provided', async () => {
+    try {
+      fixture.detectChanges();
+      throw 'It should have thrown an error about "ID"';
+    } catch (error) {
+      expect(error).toBe(EMPTY_ID_ERROR);
+    }
+  });
+
+  describe('only ID is provided, ', () => {
+    beforeEach(async () => {
+      testHost.testId = testId;
+    });
+    it('should throw an error when no label is provided', async () => {
+      try {
+        fixture.detectChanges();
+        throw 'It should have thrown an error about "LABEL"';
+      } catch (error) {
+        expect(error).toBe(EMPTY_LABEL_ERROR);
+      }
+    });
+  });
+
+  describe('all required properties are provided, ', () => {
+    beforeEach(async () => {
+      testHost.testId = testId;
+      testHost.testLabel = testLabel;
+      testHost.testPlaceholder = testPlaceholder;
+      testHost.testHasFailed = true;
+      dsfrPasswordHarness = await harnessLoader.getHarness<DsfrPasswordHarness>(
+        DsfrPasswordHarness
+      );
+    });
+
+    it('should have the right placeholder for the input attribute', async () => {
+      const placeholder = await dsfrPasswordHarness.getInputPlaceholderAttribute();
+      expect(placeholder).toEqual(testPlaceholder);
+    });
+
+    it('should have the right id for the input attribute', async () => {
+      const id = await dsfrPasswordHarness.getInputId();
+      expect(id).toEqual(`${testId}-input`);
+    });
+
+    it('should display the right class when the attribute hasFailed is true', async () => {
+      const classes: string | null =
+        await dsfrPasswordHarness.getContainerComponentAttribute('class');
+      expect(classes).toContain('fr-input-group--error');
+    });
```

Comment:
You can easily do the same for `hasSucceeded`

Reply:
Not treated entirely

Reply:
You actually need to move the lines `testHost.testHasFailed = true;` and `testHost.testHasSucceeded = true;` to each associated `it()`...